### PR TITLE
Fix for STM32H port - IPv6 MAC filter and build 

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes the build for STM32H port

Test Steps
-----------
Tested with HW demo.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c [hunk 1]
```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
...
+                        ucMACAddress[ 3 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 13 ];
+                        ucMACAddress[ 4 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 14 ];
+                        ucMACAddress[ 5 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 15 ];
+
+                        /* Allow traffic destined to Solicited-Node multicast address of this endpoint
+                         * for Duplicate Address Detection (DAD) */
+                        prvMACAddressConfig( &xEthHandle, xMACEntry, ucMACAddress );
+                        xMACEntry += 8;
+                    }
+                    else
```
Comment (ActoryOu) [inline source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:383]: The else (line 383) here is combined with line 394. ( `if( xMACEntry > ( BaseType_t ) ETH_MAC_ADDRESS3 )` )

I think you're using else for IPv4. So please correct the logic here.

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c [hunk 2]
```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
             HAL_ETH_DescAssignMemory( &( xEthHandle ), uxIndex, pucBuffer, NULL );
         }
 
+        #if ( ipconfigUSE_MDNS == 1 )
```
Comment (moninom1) [inline source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:336]: Should we add && ( ipconfigUSE_IPv6 != 1 )

Comment (tony-josi-aws) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:336]: No they are different MAC addresses

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c [hunk 3]
```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
...
+            }
+        #endif
+        #if ( ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_IPv6 != 0 ) )
+            {
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xMDNS_MACAdressIPv6.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ipconfigUSE_LLMNR == 1 )
+            {
```
Comment (moninom1) [inline source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:350]: Why are we going through both line 349 and line 356 code when ipconfigUSE_IPv6 != 0

Comment (tony-josi-aws) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:350]: For IPv4 and IPv6 we have different MAC addresses.

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c (hunk 1 above)
Comment (tony-josi-aws) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c]: Updated

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c (hunk 3 above)
Comment (moninom1) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:350]: Yes that is why, we should have  IPv4 check in place for first #if ipconfigUSE_LLMNR

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c (hunk 2 above)
Comment (moninom1) [reply source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:336]: Yes that is why Ipv4 check is needed.

File: source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c [hunk 4]
```diff
@@ -203,6 +203,34 @@ const PhyProperties_t xPHYProperties =
...
+{
+    uint32_t ulTempReg;
+    uint32_t ulETH_MAC_ADDR_HBASE = ( uint32_t ) &( heth->Instance->MACA0HR );
+    uint32_t ulETH_MAC_ADDR_LBASE = ( uint32_t ) &( heth->Instance->MACA0LR );
+
+    /* ETH_MAC_ADDRESS0 reserved for the primary MAC-address. */
+    configASSERT( ulIndex >= ETH_MAC_ADDRESS1 );
+
+    /* ETH_MAC_ADDRESS0 - ETH_MAC_ADDRESS3 */
+    configASSERT( ulIndex <= ETH_MAC_ADDRESS3 );
```
Comment (shubnil) [inline source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c:218]: Please add a comment also that more than 4 MAC addresses should not enabled simultaneously.

**Review by shubnil — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**